### DISCUSSION TITLE
Trouble with showing relative git repo path on Windows PowerShell

### DISCUSSION THREAD

**Discussion by neon16bit** [JanDeDobbeleer/oh-my-posh#5512] [Troubleshoot] [answered]
I would like to show relative repo path from the root of the repo when within git repo, i am doing that with:

```toml
template = "{{ if and .Segments.Git.RepoName (ne .Segments.Git.RepoName .Path) }}{{ .Segments.Git.RepoName }}{{ trimPrefix .Segments.Git.Dir .PWD }}{{ else }}{{ .Path }}{{ end }}"
```
However it doesn't seem to work on PowerShell unless I directly open the repo from file explorer, if I cd into the repo I get the reponame with fullpath but it seems to work well in wsl bash.

On Windows PowerShell:
![image](https://github.com/user-attachments/assets/e487b817-ab77-4ac0-b8b5-bac125193a7b)

on wsl bash:
![image](https://github.com/user-attachments/assets/bf68c0ce-8bda-43e4-883d-a87c6c1f4b23)

I want to make it function the same way it does on wsl bash. It doesn't seem to work on git bash either so I am assuming it's a windows problem.

**Comment by JanDeDobbeleer** [comment]
@neon16bit this is due to Windows paths and their default use of the `\` path separator. If you use the template functions to replace those with `/`, the template will also work on Windows. Git always uses `/` as a path separator, regardless of the platform.

**Comment by lewis-yeung** [comment]
@neon16bit I failed to reproduce the issue. Your template works as expected on my end.

@JanDeDobbeleer In a `git` segment, `.Dir` shows me a path with backslashes (`\`) as separators on Windows. Does it behave differently for you?

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung it might depend on where you cloned from. And I might be mistaken due to using parallels in the past where this was the case (as I cloned from macOS).

**Reply by neon16bit** [reply]
@lewis-yeung Actually I remember it working perfectly before, I think, because I use wsl so much I could be confusing between them. But I did test out the same template before and it worked. Initially I thought maybe it was because of an update and rolled back an update but that didn't work. I wonder what's up with that..

**Reply by neon16bit** [reply]
@JanDeDobbeleer  @lewis-yeung so I did some experiments, did clean installation via winget and scoop, both of these had the same issue. However the GitHub release worked but that makes no sense so I kept looking and lo and behold the path is case sensitive!

Works (the path as is):
![image](https://github.com/user-attachments/assets/61263721-8b63-49ab-9d82-278248fdf949)

Doesn't work (lower case path):
![image](https://github.com/user-attachments/assets/1d55dc8d-60df-4415-bde4-cd0139e62e81)

Doesn't work (lower case repo name):
![image](https://github.com/user-attachments/assets/af1dc0fb-4188-49ce-ae6e-acd621ac1699)

![image](https://github.com/user-attachments/assets/1c7f338c-b358-4738-9c35-260e173c6a85)

Basically, it doesn't work unless the path is exactly as is, capitalizations and all. I wonder if you can do something about it.

**Reply by JanDeDobbeleer** [reply]
@neon16bit that does imply you gave the repo a different name, right? Because out of the box a clone will respect the cases. In this case you need to lower both paths in the template, then do the replacement.

**Reply by neon16bit** [reply]
@JanDeDobbeleer no, the repo name is as is. it's the total path, if there's a directory with capitalized name you need to provide with the exact capitalizations, or it won't work. 

![image](https://github.com/user-attachments/assets/743a8dba-4294-4765-8278-84df4b0dd493)

for this example, it will only work if the path is "`h:/Repositories/UniSyncBackend`" and won't work for the following examples:
 - `h:/repositories/UniSyncBackend`
 - `h:/RepositorieS/UniSyncBackend`
 - `h:/REPOSITORIES/UniSyncBackend`
 - `h:/Repositories/unisyncbackend`
 - `h:/Repositories/UnisyncBackend`
 - `h:/Repositories/UNISYNCBACKEND`
 - `h:/repositories/unisyncbackend`
 - `h:/REPOSITORIES/UNISYNCBACKEND`
 
It's needs to be as is or it won't work.

**Reply by JanDeDobbeleer** [reply]
@neon16bit ok sure, but HOW is this inconsistent to begin with? It makes no sense.

**Reply by lewis-yeung** [reply]
I still cannot reproduce it in PowerShell v7.4.4 on Windows 11.

**Reply by lewis-yeung** [reply]
@neon16bit Oh, you're using **Windows PowerShell**. I can reproduce it in Windows PowerShell v5.1 where the `$PWD` is not automatically cleaned to the exact path like it is in PowerShell 7 (say, `h:/repositories/unisyncbackend` can be automatically cleaned to `H:/Repositories/UniSyncBackend`).

**Reply by neon16bit** [reply]
@lewis-yeung Yes, Windows PowerShell seems to be the culprit.

@JanDeDobbeleer it's not inconsistent at all, I misunderstood your reply. 

```yaml
{{ if and .Segments.Git.RepoName (ne .Segments.Git.RepoName .Path) }}{{ .Segments.Git.RepoName }}{{ trimPrefix (lower .Segments.Git.Dir) (lower .PWD) }}{{ else }}{{ .Path }}{{ end }}
```
This workaround fixes the issue on Windows PowerShell, but it makes all paths inside a Git repository lowercase. Thank You!

**Reply by lewis-yeung** [reply, marked as answer]
@neon16bit The following template preserves cases of the path used for the `cd` command in Windows PowerShell:

```
{{ if .Segments.Git.RepoName }}{{ .Segments.Git.RepoName }}{{ if hasPrefix (lower .Segments.Git.Dir) (lower .PWD) }}{{ substr (len .Segments.Git.Dir) (len .PWD) .PWD }}{{end}}{{ else }}{{ .Path }}{{ end }}
```

**Reply by neon16bit** [reply]
@lewis-yeung Thanks!
